client/highlevel-rest/src/main/java/org/elasticsearch/client/SearchHit.java

```diff
@@ -21,185 +21,97 @@
 
-    private Map<String, Object> hit;
-    private XContentAccessor objectPath;
+    private final XContentAccessor objectPath;
 
-    public ClientSearchHit(Map<String, Object> hit) {
-        this.hit = hit;
-        this.objectPath = new XContentAccessor(this.hit);
-    }
-
-    @Override
-    public void readFrom(StreamInput in) throws IOException {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public void writeTo(StreamOutput out) throws IOException {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
-        builder.map(this.hit);
-        builder.endObject();
-        return builder;
+    public SearchHit(Map<String, Object> hit) {
+        this.objectPath = new XContentAccessor(hit);
     }
 
     @Override
     public Iterator<SearchHitField> iterator() {
-        return fields().values().iterator();
-    }
-
-    @Override
-    public float score() {
-        return this.objectPath.evaluateDouble("_score").floatValue();
+        return getFields().values().iterator();
     }
 
-    @Override
     public float getScore() {
-        return score();
-    }
-
-    @Override
-    public String index() {
-        return this.objectPath.evaluateString("_index");
+        return this.objectPath.evaluateDouble("_score").floatValue();
     }
 
-    @Override
     public String getIndex() {
-        return index();
-    }
-
-    @Override
-    public String id() {
-        return this.objectPath.evaluateString("_id");
+        return this.objectPath.evaluateString("_index");
     }
 
-    @Override
     public String getId() {
-        return id();
-    }
-
-    @Override
-    public String type() {
-        return this.objectPath.evaluateString("_type");
+        return this.objectPath.evaluateString("_id");
     }
 
-    @Override
     public String getType() {
-        return type();
+        return this.objectPath.evaluateString("_type");
     }
 
-    @Override
     public NestedIdentity getNestedIdentity() {
         // TODO
         return null;
     }
 
-    @Override
-    public long version() {
+    public long getVersion() {
         Long version = this.objectPath.evaluateLong("_version");
         if (version == null) {
             return -1L; // same as returned by InternalSearchHit if version not set
         }
         return version;
     }
 
-    @Override
-    public long getVersion() {
-        return version();
-    }
-
-    @Override
-    public BytesReference sourceRef() {
-        return new BytesArray(sourceAsString());
-    }
-
-    @Override
-    public BytesReference getSourceRef() {
-        return sourceRef();
-    }
-
-    @Override
-    public byte[] source() {
-        return BytesReference.toBytes(sourceRef());
-    }
-
-    @Override
+    //TODO add hasElement support to XContentAccessor?
     public boolean hasSource() {
```

Comment:
I'd rather only propose `getSource()` that returns null when no response body, an empty map when the response body is empty and a non-empty map that reflects the response body.